### PR TITLE
Generalized Reed Solomon, BCH, q-BCH

### PR DESCRIPTION
**Pull request by MJunaidAftab**
Here are my submissions. In all the 3 files, I have commented out the relations section. I am getting the error`Additional properties are not allowed ('children' was unexpected)`. Not sure why this is the case? Do I need to create additional nodes/files?

### CODE REVIEW

File: codes/classical/bits/bch.yml [hunk 1]
```diff
@@ -8,21 +8,28 @@ physical: bits
...
-    - code_id: reed_solomon
-      detail: ''
-    - code_id: cyclic
-      detail: ''
-    - code_id: goppa
-      detail: ''
-# MacWilliams Sloane Ch. 12
+introduced: '\cite{doi:10.1016/S0019-9958(60)90287-4}'
+
+description: 'Bose–Chaudhuri–Hocquenghem (BCH) codes can be defined as subfield codes of the Reed-Solomon codes. If \( C_{\text{RS}} \) is a Reed-Solomon code containing elements in the finite field \( \text{GF}(2^{m}) \) for some \(m \geq 0 \), then the corresponding BCH code is defined as the code \( C_{\text{BCH}} = C_{\text{RS}} \cap (\text{GF}(2))^n \). That is, the BCH codes contains only those Reed-Solomon codewords whose components are elements of \( \text{GF}(2) \).'
```
Comment (AlexanderBarg) [inline codes/classical/bits/bch.yml:13]: This is not fully accurate. q-ary BCH codes of length n=q^m -1 can be defined as subfield subcodes of q^m-ary RS codes of length q^m-1. 

A more traditional definition of a BCH code is as follows. A q-ary BCH code of length n=q^m-1 with designed distance d is a cyclic code with zeros alpha, alpha^2,...,alpha^d-1, where alpha is a primitive element of the field q^m-1.

File: codes/classical/bits/bch.yml [hunk 2]
```diff
@@ -8,21 +8,28 @@ physical: bits
...
+
+features:
+
+    decoders:
+
+        - 'The decoding process is similar to Reed-Solomon codes. In particular, the Berlekamp-Massey algorithm can be used to decode BCH codes.'
+
+
+notes:
+  - 'BCH codes are examples of cyclic codes. This is clear since the conventional Reed-Solomon Codes are cyclic codes.'
```
Comment (AlexanderBarg) [inline codes/classical/bits/bch.yml:24]: Remove this note (the term "conventional RS code" is nonstandard, and the statement that they are cyclic is not fully correct).

File: codes/classical/q-ary_digits/generalized_reed_solomon.yml [hunk 3]
```diff
@@ -7,26 +7,34 @@ code_id: generalized_reed_solomon
...
+
+realizations:
+  - 'Commonly used in mass storage systems such as CDs and DVDs and QR codes etc.'
+
+notes:
+  - 'Generalized Reed Solomon codes contain the special case of conventiontal Reed Solomon codes. These are cyclic codes, and can also be defined by using the discrete Fourier transform. '
+
+#relations:
+#  children:
+#    - code_id: reed_solomon
```
Comment (AlexanderBarg) [inline codes/classical/q-ary_digits/generalized_reed_solomon.yml:34]: MDS codes are parents, not cousins

File: codes/classical/bits/bch.yml (hunk 2 above)
Comment (MJunaidAftab) [reply codes/classical/bits/bch.yml]: Done.

File: codes/classical/q-ary_digits/generalized_reed_solomon.yml [hunk 4]
```diff
@@ -7,26 +7,34 @@ code_id: generalized_reed_solomon
...
-  cousins:
-    - code_id: goppa
-    - code_id: bch
+name: 'Generalized Reed Solomon (GRS) code'
+introduced: '\cite{doi:10.1137/0108018}'
+
+description: 'Generalized Reed-Solomon codes are an infinite family of algebraic codes defined over the finite field \( F = GF(p^m)\) for \(m \geq 0 \) and prime \( p \). The parameters of the code are \( [n,k,n-k+1] \) where \(n \leq 2^m-1\), the number of non-zero elements in the finite field. The algebraic code exploits the bijective correspondence between \(F^k\), the \(k\)-dimensional vector space over \(F\) and \(F_k[x]\), the polynomial ring over \(F\) consisting of univariate polynomials of degree at most \(k  - 1\). Let \( (\alpha_1, \cdots, \alpha_n) \) be \( n \) distinct non-zero elements of \(F\), and let \( (v_1, \cdots, \v_n) \) be defined similarly. The message \( (c_0, \cdots, c_{k-1}) \), which can be represented as the polynomial \( p(x) = c_0 + c_1 x + \cdots + c_{k-1} x^{k-1} \), is encoded as the codeword \( ( v_1 f(\alpha_1), \cdots, v_n f(\alpha_n)) \).'
+
+
+protection: 'The code can detect \( n-k \) bit errors, and can correct  errors \( \floor {\frac{n-k}{2}} \) bit errors.'
```
[the remark this answers is not in the document]
Comment (MJunaidAftab) [reply codes/classical/q-ary_digits/generalized_reed_solomon.yml]: Done.

File: codes/classical/q-ary_digits/generalized_reed_solomon.yml (hunk 3 above)
Comment (MJunaidAftab) [reply codes/classical/q-ary_digits/generalized_reed_solomon.yml:37]: Done.

File: codes/classical/bits/bch.yml (hunk 1 above)
Comment (MJunaidAftab) [reply codes/classical/bits/bch.yml:13]: I'll use the definition of BCH codes you've mentioned.

File: codes/classical/q-ary_digits/q-bch.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+code_id: q-bch
+physical: q-ary_digits
+logical: q-ary_digits
+
+name: 'Bose–Chaudhuri–Hocquenghem (BCH) code'
+introduced: '\cite{doi:10.1016/S0019-9958(60)90287-4}'
+
+description: 'Bose–Chaudhuri–Hocquenghem (BCH) codes can be defined as subfield codes of the Reed-Solomon codes. If \( C_{\text{RS}} \) is a Reed-Solomon code containing elements in the finite field \( \text{GF}(p^{m}) \) for some prime \( p \) and \(m \geq 0 \), then the corresponding BCH code is defined as the code \( C_{\text{BCH}} = C_{\text{RS}} \cap (\text{GF}(p))^n \). That is, the BCH codes contains only those Reed-Solomon codewords whose components are elements of \( \text{GF}(p) \).'
```
[the remark this answers is not in the document]
Comment (MJunaidAftab) [reply codes/classical/q-ary_digits/q-bch.yml]: I'll add the traditional definition.

**Comment by valbert4** [comment]
> Here are my submissions. In all the 3 files, I have commented out the relations section. I am getting the error`Additional properties are not allowed ('children' was unexpected)`. Not sure why this is the case? Do I need to create additional nodes/files?

Thanks for the submissions. You can only put parents. For the children, go to their code files and put this code as parent.

File: codes/classical/bits/bch.yml [hunk 6]
```diff
@@ -8,21 +8,22 @@ physical: bits
 logical: bits
 
 name: 'Bose–Chaudhuri–Hocquenghem (BCH) code'
-short_name: 'BCH'
-introduced: '\cite{doi:10.1016/s0019-9958(60)90287-4}'
+introduced: '\cite{doi:10.1016/S0019-9958(60)90287-4}'
```
Comment (valbert4) [inline codes/classical/bits/bch.yml:11]: Also \cite{manual:{A. Hocquenghem. Codes correcteurs d'erreurs. Chiffres (Paris), 2:147–156, September 1959.}}

File: codes/classical/bits/bch.yml [hunk 7]
```diff
@@ -8,21 +8,22 @@ physical: bits
...
-    - code_id: reed_solomon
-      detail: ''
-    - code_id: cyclic
-      detail: ''
-    - code_id: goppa
-      detail: ''
-# MacWilliams Sloane Ch. 12
+        - 'The decoding process is similar to Reed-Solomon codes. In particular, the Berlekamp-Massey algorithm can be used to decode BCH codes.'
+
+#relations:
```
Comment (valbert4) [inline codes/classical/bits/bch.yml:21]: keep parent same

File: codes/classical/bits/bch.yml [hunk 8]
```diff
@@ -8,21 +8,22 @@ physical: bits
...
 
-  cousins:
-    - code_id: reed_solomon
-      detail: ''
-    - code_id: cyclic
-      detail: ''
-    - code_id: goppa
-      detail: ''
-# MacWilliams Sloane Ch. 12
+        - 'The decoding process is similar to Reed-Solomon codes. In particular, the Berlekamp-Massey algorithm can be used to decode BCH codes.'
```
Comment (valbert4) [inline codes/classical/bits/bch.yml:19]: ref for decoder

**Pull request closed without merging**